src/GameData.ts

```diff
@@ -0,0 +1,292 @@
+        if (!search) {
+            return `There are currently ${this.itemData.length} items in my lookup data!`;
+        }
+
+        const result = this.itemData.filter(c => Math.min(
+            ...[
+                levenshteinDistance(search, c.id.toString()),
+                levenshteinDistance(search, c.name),
+                levenshteinDistance(search, c.categories.toString()),
+            ]) < 1);
+
+        if (result.length === 0) {
+            return "Unable to find any data for that search term.";
+        }
+
+        if (result.length > 3) {
+            let response = "Too many results returned for that search, please try another;\n";
+            result.forEach(r => response += `\`${r.name}\`, `);
+            response = response.slice(0, -2);
+            return response;
+        }
+
+        const returnedValue = {
+            id: result[0].id,
+            name: result[0].name,
+            combineCost: result[0].price,
+            cost: result[0].priceTotal,
+            from: result[0].from,
+            to: result[0].to,
+        } as any;
+
+        return "```" + JSON.stringify(returnedValue, null, 4) + "```";
+    }
+
+    public findPerk(search: string): string {
+        if (!search) {
+            return `There are currently ${this.perkData.length} perks in my lookup data!`;
+        }
+
+        const result = this.perkData.filter(c => Math.min(
+            ...[
+                levenshteinDistance(search, c.id.toString()),
+                levenshteinDistance(search, c.name),
+                levenshteinDistance(search, c.shortDesc),
+            ]) < 1);
+
+        if (result.length === 0) {
+            return "Unable to find any data for that search term.";
+        }
+
+        if (result.length > 1) {
+            let response = "Too many results returned for that search, please try another;\n";
+            result.forEach(r => response += `\`${r}\`, `);
+            response = response.slice(0, -2);
+            return response;
+        }
+
+        const returnedValue = {
+            id: result[0].id,
+            name: result[0].name,
+            endOfGameStatDescs: result[0].endOfGameStatDescs,
+        } as any;
+
+        return "```" + JSON.stringify(returnedValue, null, 4) + "```";
+    }
+
+    public findChampion(search: string): string {
+        if (!search) {
+            return `There are currently ${this.champData.length} champions in my lookup data!`;
+        }
+
+        const result = this.champData.filter(c => Math.min(
+            ...[
+                levenshteinDistance(search, c.id.toString()),
+                levenshteinDistance(search, c.name),
+                levenshteinDistance(search, c.key),
+            ]) < 3);
+
+        if (result.length === 0) {
+            return "Unable to find any data for that search term.";
+        }
+
+        if (result.length > 1) {
+            let response = "Too many results returned for that search, please try another;\n";
```

Comment:
; in this string?

Reply:
Ment as a list separator, but apparently you are supposed to use : if the list has more than 3 elements, so its changed now